lib/hotel.rb

```diff
@@ -0,0 +1,93 @@
+require_relative "room"
+require_relative "reservation"
+require_relative "block"
+
+module HotelSystem
+  class Hotel
+    attr_reader :all_rooms
+    attr_accessor :all_reservations, :all_blocks
+
+    def initialize
+      @all_reservations = []
+      @all_rooms = []
+      @all_blocks = []
+      generate_rooms(20)
+    end
+
+    def generate_rooms(number_of_rooms)
+      number_of_rooms.times do |i|
+        @all_rooms << Room.new(number: i + 1)
+      end
+    end
+
+    def see_reservations_by_date(date)
+      this_dates_reservations = []
+      @all_reservations.each do |reservation|
+        this_dates_reservations << reservation if reservation.start_date == date
+      end
+      return this_dates_reservations
+    end
+
+    def see_available_rooms_by_date(possible_start_date, possible_end_date)
+      these_dates_available_rooms = []
+      @all_rooms.each do |room|
+        these_dates_available_rooms << room if room.date_available?(possible_start_date, possible_end_date)
```

Comment:
On line 34, I like that you call `room.date_available?` instead of having a bunch of date math here in `Hotel`. This is what Metz is talking about when she says you should "ask for _what_ instead of telling _how_".